

---

# 📚 New Real-life Example:

A factory produces metal rods.  
You collect a **sample of 50 rods** and measure their **weights** (in kilograms).

You find:

- Sample mean weight = **5.2 kg**
- Known **population σ = 0.3 kg**
- Sample standard deviation (from sample) = **0.32 kg**
- Sample size = **50 rods**
- Confidence level = **95%**

---

# 🔵 CASE 1: **σ is Known** (Use Normal Distribution)

Use `stats.norm.interval()`

---

In [1]:
from scipy import stats
import numpy as np

# Given
mean_weight = 5.2
sigma = 0.3  # known population std dev
n = 50
confidence = 0.95

# Standard error
standard_error = sigma / np.sqrt(n)

# Confidence interval using normal distribution
ci_norm = stats.norm.interval(confidence,mean_weight,standard_error)

print(f"[Known σ] 95% Confidence Interval: {ci_norm[0]:.3f} kg to {ci_norm[1]:.3f} kg")

[Known σ] 95% Confidence Interval: 5.117 kg to 5.283 kg


In [2]:
ci_norm

(5.11684577053902, 5.283154229460981)

# 🟢 CASE 2: **σ is Unknown** (Use Student's T-distribution)

Use `stats.t.interval()`


In [3]:
# Sample standard deviation
sample_std = 0.32  # from sample

# Standard error
standard_error_sample = sample_std / np.sqrt(n)

# Degrees of freedom
df = n - 1

# Confidence interval using t-distribution
ci_t = stats.t.interval(confidence, df=df, loc=mean_weight, scale=standard_error_sample)

print(f"[Unknown σ] 95% Confidence Interval: {ci_t[0]:.3f} kg to {ci_t[1]:.3f} kg")


[Unknown σ] 95% Confidence Interval: 5.109 kg to 5.291 kg


---

# 🔥 Example Output:

```
[Known σ] 95% Confidence Interval: 5.117 kg to 5.283 kg
[Unknown σ] 95% Confidence Interval: 5.109 kg to 5.291 kg
```

✅ Notice:
- The **T-interval** is slightly **wider** than the **Normal interval** because the sample adds uncertainty.
- The **difference becomes smaller** when **n is large** (like 50), but is huge if **n is small**.

---

# 📋 Recap:

| Situation                        | Function               | Interval Example |
|:----------------------------------|:----------------------:|:-----------------:|
| σ known, use Z (Normal)           | `stats.norm.interval()` | 5.117 kg to 5.283 kg    |
| σ unknown, use T (Student's t)    | `stats.t.interval()`    | 5.109 kg to 5.291 kg    |

---

